Fix legacy algorithm for <semantics> and remove obsolete attributes (#20347)
mdn/content: files/en-us/web/mathml/element/semantics/index.md

>
> - If no other rules apply: By default only the first child is rendered, which is supposed to be Presentation MathML.
> - If the first child is a Presentation MathML element other than `<annotation>` or `<annotation-xml>`, render the first child.

> > - If no Presentation MathML is found, render the first `<annotation>` or `<annotation-xml>` child element of `<semantics>` without an `src` attribute, with the additional requirement for `<annotation-xml>` elements that the `encoding` attribute is equal to one of following values:

Reviewer: ```suggestion
> - If no Presentation MathML is found, render the first `<annotation>` or `<annotation-xml>` child element of `<semantics>` without a `src` attribute. For `<annotation-xml>` elements the `encoding` attribute must be equal to one of following values:
```
